shor/transpilers/qiskit.py

```diff
@@ -6,9 +6,11 @@
 
 
 def to_qiskit_circuit(shor_circuit: QC) -> QuantumCircuit:
+    qbit_registers, cbit_registers = shor_circuit.to_registers()
+
     qiskit_circuit = QuantumCircuit(
-        len(shor_circuit.initial_state()),
-        len(shor_circuit.measure_bits()),
+        len(qbit_registers[0]),
```

Comment:
This is super close, but I think some unit tests will be failing here.

Let's write some multi qbit, multi cbit unit tests and make sure those pass.

qbits1 = Qbits(1)
qbits2 = Qbits(2)
X = QC()
X.add(qbits1)
X.add(qbits2)
X.add(H(qbits1[0]))
X.add(X(qbits2[1]))
X.add(Measure(qbits1[0], qbits2[1], CBits(2))

result = X.run(100)

Expect
result.counts ~ {'01': 50, '11' 50}